src/Microsoft.AspNetCore.Routing/Matching/JumpTableBuilder.cs

```diff
@@ -84,11 +84,7 @@ public static JumpTable Build(int defaultDestination, int exitDestination, (stri
                 fallback = new DictionaryJumpTable(defaultDestination, exitDestination, pathEntries);
             }
 
-#if IL_EMIT
             return new ILEmitTrieJumpTable(defaultDestination, exitDestination, pathEntries, vectorize: null, fallback);
-#else
-            return fallback;
```

Comment:
Off topic:

@rynowak What do you think about testing whether IL emit is supported here? 3.0 will add `RuntimeFeature.IsDynamicCodeSupported`

Reply:
We should be checking RuntimeFeature.IsDynamicCodeCompiled no?

Reply:
Sure, since we're doing this for perf that one would be better